include/gz/sensors/CameraSensor.hh

```diff
@@ -146,6 +146,10 @@ namespace gz
       /// \todo(iche033) Make this function virtual on Harmonic
       public: bool HasInfoConnections() const;
 
+      /// \brief Get the camera optical frame
+      /// \return The camera optical frame
+      public: std::string OpticalFrameId() const;
```

Comment:
```suggestion
      public: const std::string &OpticalFrameId() const;
```

Reply:
I originally did this but the rest of the API returns string by value like `FrameId()`.

Reply:
@azeey Let me know if I should still make the change.

Reply:
Not sure if the extra copy makes sense especially since this will be called every time a message is published. I'd vote for const ref. Maybe @iche033 can weigh in.

Reply:
yeah unfortunately it breaks API consistency but let's go with `const &` since as it makes sense here, and we'll start doing this more in new APIs.

Reply:
I updated to return `const&`